static/js/dark_theme.js

```diff
@@ -24,3 +25,11 @@ export function disable() {
 export function default_preference_checker() {
     $("body").removeClass("dark-theme").addClass("color-scheme-automatic");
 }
+
+// Rerender the realm-logo when the browser detects color scheme changes.
+const dark_mode_media_query_list = window.matchMedia("(prefers-color-scheme: dark)");
+dark_mode_media_query_list.addEventListener("change", () => {
+    if ($("body").hasClass("color-scheme-automatic")) {
+        realm_logo.render();
+    }
+});
```

Comment:
I think this code belongs in the `realm_logo` module; and it should be inside an `initialize` method; we try to avoid running anything on import as it's better for unit testing.

I think the other detail is we probably want this to only run if it's been rendered at least once before; you might want a module-level global to keep track of that.